fix: HAIP interoperability for x509_hash (#884)
italia/eidas-it-wallet-docs: docs/en/remote-flow.rst

  4. *WI Checks*: the Wallet Instance:

    a. verifies the signature of the signed Request Object using the public key identified in the ``JWT`` header of the Request Object. Using that reference, the Wallet Instance is able to select the correct Relying Party's public key for signature verification (:ref:`WP_085 <wallet-credential-presentation-testcases>`).
    b. verifies that the ``client_id`` contained in the Request Object issuer (Relying Party) matches with the one obtained at the step number 2:
    
       * If ``client_id`` uses the ``openid_federation`` prefix, it MUST match the ``sub`` parameter contained in the Relying Party's Entity Configuration within the Trust Chain (:ref:`WP_086 <wallet-credential-presentation-testcases>`).
       * If ``client_id`` uses the ``x509_hash`` prefix, the Wallet Instance MUST verify that the hash of the Relying Party’s X.509 certificate (in the ``x5c`` request header) matches the hash contained in ``client_id`` from step 2 (as defined in `OpenID4VP`_, Section 5.9.3).

>     c. evaluates the requested Digital Credentials and checks the eligibility of the Relying Party in asking for these by applying the policies related to that specific Relying Party, obtained with the Trust Chain (:ref:`WP_087 <wallet-credential-presentation-testcases>`).

Reviewer: that's ok for this release. For the next milestone we'll implement RP's policies according to ARF's STS, as in review in weBuild LSP, see: https://github.com/webuild-consortium/wp4-trust-group/pull/12/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R125

    - Key ID of the public key needed to verify the JWT signature, as defined in [:rfc:`7517`]. REQUIRED when ``trust_chain`` is used.
  * - **trust_chain**

>     - Sequence of Entity Statements that composes the Trust Chain related to the Relying Party, as defined in `OID-FED`_ Section 4.3 *Trust Chain Header Parameter*.

Reviewer: ```suggestion
    - CONDITIONAL. REQUIRED when the ``client_id`` prefix used in the request is set with ``openid-federation``. It is a sequence of Entity Statements that composes the Trust Chain related to the Relying Party, as defined in `OID-FED`_ Section 4.3 *Trust Chain Header Parameter*.
```

  * - **x5c**

>     - CONDITIONAL. REQUIRED when ``client_id`` uses the ``x509_hash`` prefix; otherwise ``kid`` with ``jwks`` is used. Contains the Relying Party’s leaf X.509 certificate (and optionally intermediate certificates), used to verify the JWT signature with the public key in the Relying Party’s certificate as defined in :rfc:`7515`. The Relying Party’s certificate in ``x5c`` MUST assert identity information that corresponds to the endpoint identities conveyed in the Authorization request and response (e.g., the host of each ``response_uri`` and ``redirect_uri``). See Section 5 of `OPENID4VC-HAIP`_.

Reviewer: ```suggestion
    - CONDITIONAL. REQUIRED when ``client_id`` uses the ``x509_hash`` prefix; otherwise ``kid`` with ``jwks`` is used. Contains the Relying Party’s leaf X.509 certificate (and optionally intermediate certificates), used to verify the JWT signature with the public key in the Relying Party’s certificate as defined in :rfc:`7515`. The Relying Party’s certificate in ``x5c`` MUST assert Relying Party identity information. When the Relying Party's X.509 certificate contains X.509 SAN URI parameters, the endpoints used in the presentation request MUST be validated against X.509 SAN URI parameters value, such as ``response_uri`` and ``redirect_uri``. See Section 5 of `OPENID4VC-HAIP`_.
```

    - **Description**
  * - **client_id**
    - Unique Identifier of the Relying Party.
  * - **client_metadata**
    - A JSON object containing the Relying Party metadata values, that SHOULD include the following parameters:
        - **vp_formats_supported**. Used by the Wallet Instance to determine the supported Verifiable Presentation formats.
        - **encrypted_response_enc_values_supported**. JSON array listing the supported JWE ``enc`` algorithms for encrypted Authorization Responses in ``direct_post.jwt``.
        - **jwks**. JSON Web Key Set used by the Wallet Instance for encrypting the Authorization Response or for key agreement. Keys contained in this set are request-specific and identified by their ``kid`` value.
        - **request_uris**. JSON Array of pre-registered request_uri endpoints where the Wallet Instance may obtain a new Request Object. URLs MUST use the ``https`` scheme.

>         - **response_uris**. JSON Array of endpoints to which the Wallet Instance sends the Authorization Response using ``direct_post.jwt``. URLs MUST use the ``https`` scheme.

Reviewer: ```suggestion
```
